### PR TITLE
COMMERCE-11457 - Technical Task - Supplier Order Shipments to feed back to Original Customer Order

### PR DESCRIPTION
**Pull request by ghost**
https://issues.liferay.com/browse/COMMERCE-11280

### CODE REVIEW

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

File: modules/apps/commerce/commerce-service/service.xml [hunk 1]
```diff
@@ -402,6 +402,9 @@
 		<finder name="CProductId" return-type="Collection">
 			<finder-column name="CProductId" />
 		</finder>
+		<finder name="CustomerCommerceOrderItemId" return-type="Collection">
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-service/service.xml:405]: wrong commit name, you are adding a finder not a column, that one was already added before

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderItemServiceImpl.java [hunk 2]
```diff
@@ -341,6 +341,25 @@ public int getCommerceOrderItemsQuantity(long commerceOrderId)
...
+	@Override
+	public List<CommerceOrderItem> getSupplierCommerceOrderItems(
+			long customerCommerceOrderItemId, int start, int end)
+		throws PortalException {
+
+		CommerceOrderItem commerceOrderItem =
+			commerceOrderItemLocalService.fetchCommerceOrderItem(
+				customerCommerceOrderItemId);
+
+		if (commerceOrderItem != null) {
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderItemServiceImpl.java:353]: what if commerceOrderItem is null?

Comment (CrescenzoRegaLF) [reply modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderItemServiceImpl.java:353]: the value of 'customerCommerceOrderItemId' is passed as parameter in the GET method, if they pass a value that is not present the fetch returns null

Comment (CrescenzoRegaLF) [reply modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderItemServiceImpl.java:353]: which should return an empty list, but at this point it makes no sense to make the second query

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Review by luca-pellizzon — APPROVED** [review]

**Comment by 22LucaGiannotta22** [comment]
ci:test:commerce

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Comment by CrescenzoRegaLF** [comment]
ci:stop

**Comment by CrescenzoRegaLF** [comment]
ci:stop:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:stop:commerce

**Pull request closed without merging**